app/fan.py

```diff
@@ -94,135 +116,80 @@ def control():
     log_status(cpu_temp, nvme_temp, hdd_temp, pwm, profile)
     print(f"Applied profile '{profile}' → PWM: {pwm}")
 
+
 def status():
     cpu = read_temp_cpu()
     nvme = read_temp_nvme()
     hdd = read_temp_hdd()
     print(f"CPU Temp: {cpu}°C | NVMe Temp: {nvme}°C | HDD Temp: {hdd}°C")
 
-if __name__ == "__main__":
-    import sys
-    if len(sys.argv) < 2:
-        print("Usage: fan.py [status|control|set-profile <name>|get-profile]")
-        exit(1)
 
-    cmd = sys.argv[1]
-    if cmd == "status":
-        status()
-    elif cmd == "control":
-        control()
-    elif cmd == "set-profile" and len(sys.argv) == 3:
-        set_profile(sys.argv[2])
-    elif cmd == "get-profile":
-        get_profile()
-    else:
-        print("Unknown command.")
-import os
-import json
-import subprocess
-from datetime import datetime
-
-HWMON_PATH = "/sys/class/hwmon/hwmon4"
-PROFILE_FILE = "fan_profile.conf"
-LOG_FILE = "logs/fan.log"
-PWM_HEADERS = [1, 2, 3, 4]
-
-def read_temp_cpu():
+def list_all_drives():
+    """Return a list of all drive device paths using smartctl --scan."""
     try:
-        out = subprocess.check_output(["sensors"], encoding="utf-8")
-        for line in out.splitlines():
-            if "Package id 0" in line:
-                return int(float(line.split()[3].replace("+", "").replace("°C", "")))
-    except:
-        return 0
+        scan_out = subprocess.check_output(["smartctl", "--scan"], encoding="utf-8")
```

Comment:
btw smartclt can output json...

Reply:
but scanning disks every time is not great, should happen only one time per container startup. I dont think disks change so often...

Also it seems that this file is called run via subprocess in server.py. Why not just import the funcs and call directly?